blog/content/post/2022-12-22-fobnail_application.md

```diff
@@ -0,0 +1,357 @@
+
+- `fobnail-mount.service` is to be installed in `/lib/systemd/system` or another
+directory searched by `systemd`. It contains paths to files listed below, you
+can change it in the service file or use defaults.
+- `fobnail.cfg` by default is expected to be located in `/etc` directory. It
+holds paths and filenames used by Attester, change those if required. Note that
+it should use the same values as were used during provisioning, unless any of
+files pointet to by configuration were manually moved.
+- `mount.sh` and `umount.sh` should be installed in `/usr/share/fobnail` by
+default, create this directory if it doesn't exist.
+- `99-fobnail.rules` is the file that instructs `udev` (which is now part of
+`systemd`) to automatically start the service when Token is plugged in to the
+platform. Copy this file to `/etc/udev/rules.d` directory.
+- `10-fobnail.link` was already created before platform provisioning. This file
+doesn't have configurable location, it always have to live in
+`/etc/systemd/network` directory. Note that key used in `[Match]` section
+requires `systemd` in version 243 or newer. If you want to use it with older
+`systemd` you have to use another key, otherwise this file will be matched by
+_every_ link, even `loopback` interface. This would most likely break your
+Internet connection. Refer to your version of `man systemd.link` for alternative
+keys for `[Match]` if you want to use it with older `systemd`.
+
+To install all mentioned files in their default location with proper permissions
+go to `fobnail-attester/scripts` directory and execute in terminal:
+
+```bash
+sudo install -m 644 fobnail-mount.service -t /lib/systemd/system/
+sudo install -m 644 fobnail.cfg /etc/
+sudo install -D *.sh -t /usr/share/fobnail/
+sudo install -m 644 99-fobnail.rules -t /etc/udev/rules.d/
+# This one should already be done, leaving it here for completeness:
+sudo install -m 644 10-fobnail.link -t /etc/systemd/network/
+sudo install -m 755 ../bin/fobnail-attester /usr/bin/
+```
+
+Now all that's left is to restart services so changed configuration is actually
+used. This can be done by rebooting whole system or restarting them manually
+with:
+
+```bash
+sudo systemctl reload-or-restart systemd-udevd
+```
+
+#### Use
+
+Just plug in the Token, wait few seconds (up to few dozen) for green light and
+voila! You should now have access to all your secrets, mounted in directory
+pointed to by `$MNT_DIR` in `fobnail.cfg`.
+
+When the Token is pulled out, the secure disk should be automatically unmounted.
+Make sure that all applications using files on that disk are closed before
+removing the Token - if the files are still in use at that point, unmounting
+will fail and access to your secrets will still be possible. Keep in mind that
+it also applies to terminals with `MNT_DIR` or any of its sub-directories being
+their current working directory.
+
+## Demo
+
+**TBD**
```

Comment:
@krystian-hebel ?

Reply:
Demo is being edited, if everything goes as planned it should be ready tomorrow.

Reply:
Demo is ready, video added to the post.